app/src/main/java/me/yifeiyuan/flapdev/DifferActivity.java

```diff
@@ -34,49 +31,59 @@ protected void onCreate(Bundle savedInstanceState) {
 
         RecyclerView recyclerView = findViewById(R.id.rv_items);
 
-        flapAdapter = new FlapAdapter();
-
-        List<SimpleTextModel> models = new ArrayList<>();
-
-        models.add(new SimpleTextModel("Android"));
-        models.add(new SimpleTextModel("Java"));
-        models.add(new SimpleTextModel("Kotlin"));
-
-        flapAdapter.setData(models);
-
-        recyclerView.setAdapter(flapAdapter);
-
-        differ = new AsyncListDiffer<SimpleTextModel>(flapAdapter, new DiffUtil.ItemCallback<SimpleTextModel>() {
+        flapAdapter = new DifferFlapAdapter<>(new DiffUtil.ItemCallback<SimpleTextModel>() {
             @Override
             public boolean areItemsTheSame(@NonNull final SimpleTextModel simpleTextModel, @NonNull final SimpleTextModel t1) {
-                Log.d(TAG, "areItemsTheSame() called with: simpleTextModel = [" + simpleTextModel + "], t1 = [" + t1 + "]");
-                return false;
+                return true;
```

Comment:
`areItemsTheSame` should check whether items are the same from the perspective of business logic (e.g. if you items have unique IDs - check if IDs are equal)

Reply:
Yes , you are right . I was a test code , just wanna make it simple.